site.cfg.example

```diff
@@ -83,7 +83,7 @@
 # This is a good place to add general library and include directories like
 # /usr/local/{lib,include}
 #
-#[ALL]
+#[DEFAULT]
```

Comment:
This doesn't agree with `numpy/distutils/system_info.py`, which uses `ALL`. Why the change?

Reply:
Seems this might break people currently using a `site.cfg` file.

Reply:
If this is wrong, why are no tests failing? I think the documentation is wrong here: the `ALL` section does **not** override other section's missing values. I reverted the section name change in 8f94b292f, and added a test that shows the documentation is wrong. Do we want to fix the documentation or add code to override missing defaults?

Reply:
My only concern here is back compatibility. If the code can be changed without causing problems, that is OK. If not, I think the documentation should reflect reality. A new section, maybe `DEFAULT`, could be added if that adds useful functionality :)

Reply:
It's true that `ALL` is the default value for `system_info.section`, but all the subclasses apart from the *_opt_info ones (which do no lookups themselves) override it. So it has no effect.

The `DEFAULT` section OTOH works, because that's builtin ConfigParser functionality.

I'm not sure if the ALL stuff has worked at some point, but currently it does not, and probably the default should be changed to `section = None` in `class system_info` to avoid confusion. Also the comment on top of `system_info.py` should be corrected.

Reply:
So maybe just remove this part from site.cfg.example?

Reply:
It turns out the `*_opt_info` and `f2py_info` classes use `ALL`. The other classes all override the `system_info` class's `section` value with more specific names. I tried to document all this in b4ccd24

Reply:
The classes that use `ALL` I think don't have any tunable options.

If so, I'd suggest to not document it, and just schedule itfor removal.

Reply:
Sorry, this is not fully correct --- the `extra_compile_args` and `extra_link_args` in the `ALL` section may have an effect because they are added by `calc_extra_info`. The library variables don't have an effect, as `libraries` is empty list in `calc_libraries_info`.

Reply:
thanks for the deep dive. I will remove ALL from the docstring.

Reply:
Setting [DEFAULT] libs and headers to a location different from OpenBLAS seems to break using a Debian 10 system OpenBLAS on a Python+numpy+scipy built from source in a non-standard location. Namely, if in the following `site.cfg`
```
[ALL]
library_dirs = /home/dimpase/sage-src/local/lib
include_dirs = /home/dimpase/sage-src/local/include
[blas]
include_dirs = /usr/include/x86_64-linux-gnu
blas_libs    = openblas
[lapack]
lapack_libs  = openblas
```
one adds after [ALL] section
```
[DEFAULT]
library_dirs = /home/dimpase/sage-src/local/lib
include_dirs = /home/dimpase/sage-src/local/include
```
then scipy cannot be built any more, as it cannot find blas/lapack (and it can without [DEFAULT]  section).

This is from https://trac.sagemath.org/ticket/29051 and https://trac.sagemath.org/ticket/29051

Reply:
That's because setting DEFAULT overrides the default search paths, which *includes* `/usr/lib` etc standard places.

Reply:
> As noted elsewhere, the search paths in ALL do not do anything at all so there's no point setting them.

OK, so that's very good to know, meaning that we may skip [ALL] and [DEFAULT] for the case at hand, at least.